include/openthread/thread.h

```diff
@@ -380,30 +380,52 @@ otError otThreadSetLinkMode(otInstance *aInstance, otLinkModeConfig aConfig);
 /**
  * Get the thrNetworkKey.
  *
+ * This function will return an union of key reference and literal key. These feilds
+ * will be populated based on the Crypto module being used. For PSA Crypto, a reference
+ * to the thrNetworkKey is returned and not the literal key.
+ *
  * @param[in]   aInstance   A pointer to an OpenThread instance.
  *
  * @returns A pointer to a buffer containing the thrNetworkKey.
  *
- * @sa otThreadSetNetworkKey
+ * @sa otThreadSetNetworkKey otThreadCopyNetworkKey
  *
  */
 const otNetworkKey *otThreadGetNetworkKey(otInstance *aInstance);
 
+/**
+ * Copy the thrNetworkKey into passed buffer
+ *
+ * This function copies the thrNetworkKey as a literal into the output buffer and works
+ * irrespective of the Crypto module used.
```

Comment:
What if Network Key is stored in PSA Crypto as non-exportable to prevent NSPE from accessing it? Should this function simply return `OT_ERROR_FAILED`?

Reply:
We can't restrict the network key from being exported. It needs to be exportable as we need to form the TLV on reboot. If we make it non-exportable, we will need to store the TLVs in NVM.

Reply:
I understand it would require refactoring of multiple modules in OpenThread, but I would like to prepare for increasing security of products and eventually marking Network Key as non-exportable. I think it would be OK to return something like `ERROR_NOT_SUPPORTED` from this function if key value is not available due to security restrictions.

Reply:
The issue I am pointing is with the generation of the TLVs. Every time a new device joins the network the parent needs to send the joiner entrust message with Network Key TLV. This means we will need the network key as a literal string to create this TLV. There are two approaches to this, either wee create the TLV and store it in the NVM or create this TLV on every boot / key change.

I don't think storing the TLV in NVM is a good option as this defeats the purpose of using key refs, as the network key is still stored in plain text as part of the TLV. So we would need to create this TLV on every boot, this is possible only if the network key is exportable, at least in the FTDs.

Reply:
If OpenThread would not have access to Network Key value, we would need to create PSA ARoT for OpenThread that would handle TLV generation and storage inside SPE in a way it never leaves SPE. This way Network Key would not need to be exportable. This is a valid scenario for ultra-secure devices.